hypnos/src/event_handler.c

```diff
@@ -126,17 +127,21 @@ void button_pressed_isr(struct device *gpiobtn, struct gpio_callback *cb, uint32
 		bt_on();
 	}
 	gfx_update();
+	backlight_enable(true);
 }
 
 void touch_tap_isr(struct device *touch_dev, struct sensor_trigger *tap)
 {
-	backlight_enable(true);
 	k_timer_start(&display_off_timer, DISPLAY_TIMEOUT, K_NO_WAIT);
 	display_wake_up();
 	clock_increment_local_time();
 	clock_show_time();
 	battery_show_status();
+	accelerometer_show_health_data();
 	gfx_update();
+	backlight_enable(true);
+
+	sensor_sample_fetch(touch_dev);
```

Comment:
Why does this function take touch_dev as its argument?

Reply:
https://docs.zephyrproject.org/latest/reference/peripherals/sensor.html

This is the Zephyr sensor API. It triggers a call for `xxx_sample_fetch()` in the driver. 
In order to  fetch the sample of the correct driver, it would need the device driver structure =)

Reply:
Hmmm. If it were the accel driver it would make sense, but I think it’s weird that it takes the touch driver. Thanks for the docs though.

Reply:
ah true. That is a mistake. I removed it.